arch/x86/core/intel64/locore.S

```diff
@@ -820,16 +820,18 @@ IRQ(248); IRQ(249); IRQ(250); IRQ(251); IRQ(252); IRQ(253); IRQ(254); IRQ(255)
 #define	IRQ_STACK	2
 #define EXC_STACK	2
 #define BAD_STACK	2
+#define NMI_STACK	2
 #else
 #define	IRQ_STACK	1
+#define NMI_STACK	6 /* NMI stack */
```

Comment:
this doesn't work if CONFIG_X86_KPTI is enabled. You will need logic in `except` to switch to the per-cpu NMI stack manually. One of the reasons why I think this needs tests, it would be very good to somehow trigger an NMI on a CPU that is running a user thread (perhaps set a DPL of 3 on the NMI exception if CONFIG_TEST is turned on, and trigger from user mode with an `int` instruction, that might work although I am not 100% sure)

Reply:
@andrewboie, do you mean `except` in `arch/x86/core/intel64/locore.S`? I have not found how other stacks are handled there.

Reply:
yes. that code contains a manual stack switch if KPTI is turned on.

Reply:
Fixed for KPTI case also in the commit 43b0ae4d1bea9626c70a6e4218f6aaf1175fdb3d